rali/rali/source/tf_meta_data_reader.cpp

```diff
@@ -99,65 +101,65 @@ MetaDataReader::Status TFMetaDataReader::read_record(std::ifstream &file_content
     char * header_crc = new char [uint32_size];
     char * footer_crc = new char [uint32_size];
     file_contents.read(header_length, uint64_size);
-    if(!file_contents)
-        THROW("TFMetaDataReader: Error in reading TF records")
     file_contents.read(header_crc, uint32_size);
-    if(!file_contents)
-        THROW("TFMetaDataReader: Error in reading TF records")
     memcpy(&data_length, header_length, sizeof(data_length));
     memcpy(&length_crc, header_crc, sizeof(length_crc));
     if(length + data_length + 16 == file_size){
         _last_rec = true;
     }
     char *data = new char[data_length];
     file_contents.read(data,data_length);
-    if(!file_contents)
-        THROW("TFMetaDataReader: Error in reading TF records");
     tensorflow::Example single_example;
     single_example.ParseFromArray(data,data_length);
     tensorflow::Features features = single_example.features();
+//..............
+    // tensorflow::Features features = single_example.features();
+    // features.PrintDebugString();
+//..............
     auto feature = features.feature();
     tensorflow::Feature single_feature;
-    single_feature = feature.at("image/filename");
+    single_feature = feature.at(user_filename_key);
     std::string fname = single_feature.bytes_list().value()[0];
     _image_name.push_back(fname);
     uint label;
-    single_feature = feature.at("image/class/label");
+    single_feature = feature.at(user_label_key);
     label = single_feature.int64_list().value()[0];
     add(fname, label);
     file_contents.read(footer_crc, sizeof(data_crc));
-    if(!file_contents)
-        THROW("TFMetaDataReader: Error in reading TF records");
     memcpy(&data_crc, footer_crc, sizeof(data_crc));
-    delete[] header_length;
-    delete[] header_crc;
-    delete[] footer_crc;
-    delete[] data;
-    return ret;
+    free(header_length);
+    free(header_crc);
+    free(footer_crc);
```

Comment:
Codacy found an issue: [Mismatching allocation and deallocation: footer_crc](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5976200)